subprojects/gst-plugins-bad/sys/msdk/gstmsdkdec.c

```diff
@@ -218,52 +214,46 @@ allocate_output_surface (GstMsdkDec * thiz)
         break;
     }
   }
-#ifndef _WIN32
-  if ((gst_buffer_pool_acquire_buffer (thiz->alloc_pool, &out_buffer, NULL))
-      != GST_FLOW_OK) {
-    GST_ERROR_OBJECT (thiz, "Failed to allocate output buffer");
-    return NULL;
-  }
 
-  mem = gst_buffer_peek_memory (out_buffer, 0);
-  msdk_surface = g_slice_new0 (GstMsdkSurface);
-  msdk_surface->from_qdata = FALSE;
+  if (thiz->use_video_memory) {
+    if ((gst_buffer_pool_acquire_buffer (thiz->alloc_pool, &out_buffer, NULL))
+        != GST_FLOW_OK) {
+      GST_ERROR_OBJECT (thiz, "Failed to allocate output buffer");
+      return NULL;
+    }
 
-  if ((mfx_surface = gst_mini_object_get_qdata (GST_MINI_OBJECT_CAST (mem),
-              GST_MSDK_FRAME_SURFACE))) {
-    msdk_surface->surface = mfx_surface;
-    msdk_surface->from_qdata = TRUE;
+    msdk_surface =
+        gst_msdk_import_to_msdk_surface (out_buffer, thiz->context, NULL,
```

Comment:
I don't think we should call gst_msdk_import_to_msdk_surface here again, a check for if the out_buffer contains qdata is enough. There are two reasons to do so,  (1) all the buffers we will use for the decoder are already wrapped in the callback Alloc function, and we give back these buffers to the alloc_pool. (2) We should avoid using other buffers allocated from the pool which are not registered in the callback Alloc as MediaSDK/oneVPL don't recognize these buffers.  Please consider keeping the original logic here if possible. Thanks

Reply:
Kept the original logic but added macros for Windows to do a gst_buffer_map